src/util/proxy.ts

```diff
@@ -0,0 +1,60 @@
+import { Dispatcher, ProxyAgent, setGlobalDispatcher } from "undici";
+
+import { socksDispatcher } from "fetch-socks";
+import type { SocksProxyType } from "socks/typings/common/constants.js";
+
+export function getEnvProxyUrl() {
+  if (process.env.PROXY_SERVER) {
+    return process.env.PROXY_SERVER;
+  }
+
+  // for backwards compatibility with 0.x proxy settings
+  if (process.env.PROXY_HOST && process.env.PROXY_PORT) {
+    return `http://${process.env.PROXY_HOST}:${process.env.PROXY_PORT}`;
+  }
+
+  return "";
+}
+
+export function initProxy(proxy?: string): string {
+  if (!proxy) {
+    proxy = getEnvProxyUrl();
+  }
+  if (proxy) {
+    const dispatcher = createDispatcher(proxy);
+    if (dispatcher) {
+      setGlobalDispatcher(dispatcher);
+      return proxy;
```

Comment:
Might be worth also logging the proxy being used here as an `info` message so it's clear from the logs that the traffic was proxied?

Reply:
There is a log line here that's logged once:
https://github.com/webrecorder/browsertrix-crawler/pull/589/files#diff-44d0707556db333e2610b1c8d5a2a6da872141d6ea9a52ac978271ef83841df3R239
that's basically the same thing, just in browser.ts

Reply:
Ah great, missed that!